Structural/Decorator/JsonRenderer.php

```diff
@@ -8,4 +8,16 @@ public function renderData(): string
     {
         return json_encode($this->wrapped->renderData());
     }
+
+    /**
+     * Renders data as JSON based on options passed.
+     * In terms of decorator patters, it represents additional extended behaviour.
+     *
+     * @param int $options Options to be used to encode data into JSON.
+     * @return string Rendered output
+     */
+    public function renderDataWithOptions($options = 0): string
```

Comment:
Instead of using a new method, I think it will be more easy to understand if we add an options property to configure a JsonRenderer instance and continue to use the renderData method. This way we can chain calls with another decorators forever.